installer/dev/install.cpp

```diff
@@ -10,51 +10,82 @@ using namespace Windows::System;
 
 namespace ProjectReunionInstaller {
 
-    winrt::hresult AddPackage(const Uri& packageUri)
+    HRESULT RegisterPackage(const std::wstring& packageFullName)
+    {
+        PackageManager packageManager;
+        const auto deploymentOperation{ packageManager.RegisterPackageByFullNameAsync(packageFullName, nullptr, DeploymentOptions::None) };
+        deploymentOperation.get();
+        if (deploymentOperation.Status() != AsyncStatus::Completed)
+        {
+            return static_cast<HRESULT>(deploymentOperation.ErrorCode());
```

Comment:
ErrorCode or ExtendedErrorCode?

Reply:
ErrorCode, as per the documented sample for the Deployment API. 

https://docs.microsoft.com/en-us/uwp/api/windows.management.deployment.packagemanager.addpackageasync?view=winrt-19041

Reply:
TL;DR the sample's old and not best practice. You want to return ExtendedErrorCode.

Yes, but that sample's very old, dating back to Win8. It returns the generic error code per PackageManager.AddPackageAsync() e.g. the dreaded 0x80073CF9 aka HRESULT_FROM_WIN32(ERROR_INSTALL_FAILED).

[.ExtendedErrorCode](https://docs.microsoft.com/uwp/api/windows.management.deployment.deploymentresult.extendederrorcode?view=winrt-19041) was added shortl;y after to expose more meaningful error codes -- AccessDenied, OutOfDiskSpace, and others.

When errors happen the actual error code boils up to the API but it's almost always remapped into the general 80073CF9 "Install Failed". You have to hunt down Deployment's log to determine the actual failure. Or call `.ExtendedErrorCode`. This was intentional and well intentioned in Win8 but, 20-20 hindsight, was a mistake. Alas, PackageManager can't directly fix this without compatibility issues, hence the added property.

If the installer returns `deploymentOperation.ErrorCode` all folks will know is "Install Failed". If the installer returns `deploymentOperation.ExtendedErrorCode` the actual (rich) error code will be readily known.

We should also file a sample bug to update it given current best practices.